pkg/syncer/volume_health_reconciler.go

```diff
@@ -91,16 +143,63 @@ func NewVolumeHealthReconciler(
 		svcPVCSynced:        svcPVCInformer.Informer().HasSynced,
 		svcClaimQueue:       svcClaimQueue,
 		supervisorNamespace: supervisorNamespace,
+		volumeHandleToPVs: &volumeHandleToPVs{
+			RWMutex: &sync.RWMutex{},
+			items:   make(map[string][]string),
+		},
 	}
 
 	svcPVCInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
 		AddFunc:    rc.svcAddPVC,
 		UpdateFunc: rc.svcUpdatePVC,
 	}, resyncPeriod)
 
+	tkgPVInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc:    rc.tkgAddPV,
+		UpdateFunc: rc.tkgUpdatePV,
+		DeleteFunc: rc.tkgDeletePV,
+	}, resyncPeriod)
+
 	return rc
 }
 
+func (rc *volumeHealthReconciler) tkgAddPV(obj interface{}) {
+	pv, ok := obj.(*v1.PersistentVolume)
+	if !ok {
+		return
+	}
+	if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == csitypes.Name && pv.Status.Phase == v1.VolumeBound {
+		// Add to volumeHandleToPVs
+		rc.volumeHandleToPVs.add(pv.Spec.CSI.VolumeHandle, pv.Name)
+	}
+}
+
+func (rc *volumeHealthReconciler) tkgUpdatePV(oldObj, newObj interface{}) {
```

Comment:
PV name is immutable, I think we do not `tkgUpdatePV`. Are you adding this for other PV states - `Available`?

Reply:
We need this for the case where a PV enters Bound state right? That's the only time we add to this map.